Tools/cases_generator/generate_cases.py

```diff
@@ -43,65 +96,44 @@ class Instruction(parser.InstDef):
     predicted: bool = False
 
     def __init__(self, inst: parser.InstDef):
-        super().__init__(inst.header, inst.block)
-        self.context = inst.context
+        self.inst = inst
+        self.kind = inst.kind
+        self.name = inst.name
+        self.block = inst.block
         self.always_exits = always_exits(self.block)
         self.cache_effects = [
-            effect for effect in self.inputs if isinstance(effect, parser.CacheEffect)
+            effect for effect in inst.inputs if isinstance(effect, parser.CacheEffect)
         ]
         self.cache_offset = sum(c.size for c in self.cache_effects)
         self.input_effects = [
-            effect for effect in self.inputs if isinstance(effect, parser.StackEffect)
+            effect for effect in inst.inputs if isinstance(effect, parser.StackEffect)
         ]
-        self.output_effects = self.outputs  # For consistency/completeness
+        self.output_effects = inst.outputs  # For consistency/completeness
 
-    def write(self, f: typing.TextIO, indent: str, dedent: int = 0) -> None:
+    def write(self, out: Formatter) -> None:
         """Write one instruction, sans prologue and epilogue."""
-        if dedent < 0:
-            indent += " " * -dedent  # DO WE NEED THIS?
-
-        # Get cache offset and maybe assert that it is correct
+        # Write a static assertion that a family's cache size is correct
         if family := self.family:
             if self.name == family.members[0]:
                 if cache_size := family.size:
-                    f.write(
-                        f"{indent}    static_assert({cache_size} == "
-                        f'{self.cache_offset}, "incorrect cache size");\n'
-                    )
-
-        # Write cache effect variable declarations
```

Comment:
This block has been moved into `write_body`. This move is what caused the one change in generated code.